[FSDP()][5/N] Refactor `MixedPrecision` dataclass

awgu opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack):
* #87942 [FSDP()][27/N] Add forward hook registration
* #87941 [FSDP()][26/N] Move `_lazy_init()` into `_fsdp_root_pre_forward()`
* #87940 [FSDP()][25/N] Add `_post_forward_reshard()`
* #87939 [FSDP()][24/N] Refactor `_lazy_init()`
* #87938 [FSDP()][23/N] Refactor handle attr initialization
* #87937 [FSDP] Simplify `_reset_lazy_init()`
* #87936 [FSDP()][22/N] Refactor `_cast_buffers()` in `_lazy_init()`
* #87935 [FSDP()][21/N] Refactor `_buffer_name_to_orig_dtype` computation
* #87934 [FSDP] Rename `dtype` to `buffer_name_to_dtype`
* #87933 [FSDP] Remove `device` arg from `_cast_buffers()`
* #87932 [FSDP()][20/N][Easy] Move functions in file
* #87931 [FSDP()][18/N] Refactor `pre_forward_unshard()`
* #87930 [FSDP()][17/N] Refactor `_fsdp_root_pre_forward()`
* #87929 [FSDP()][16/N] Refactor post-forward/pre-backward
* #87928 [FSDP()][15/N] Refactor `_init_streams()`
* #87927 [FSDP()][14/N] Refactor pre-forward/post-backward
* #87926 [FSDP()][13/N] Refactor unshard/reshard/grads
* #87925 [FSDP()][12/N] Easy cleanup
* #87924 [FSDP()][10/N][11/N] Introduce composable (ctor only)
* #87923 [FSDP()][9/N] Refactor ctor (continued)
* #87922 [FSDP()][8/N] Refactor limiter's `_FreeEventQueue`
* #87921 [FSDP()][7/N] Refactor most of ctor
* #87920 [FSDP()][6/N] Refactor `CPUOffload` dataclass
* **#87919 [FSDP()][5/N] Refactor `MixedPrecision` dataclass**
* #87918 [FSDP()][4/N] Refactor `ShardingStrategy` enum
* #87917 [FSDP()][3/N] Refactor `BackwardPrefetch` enum

This PR is easy. I moved `MixedPrecision` to `common_utils.py` and reworded the docs a bit.

Commit:
[FSDP()][5/N] Refactor `MixedPrecision` dataclass

[ghstack-poisoned]

Commit:
Update on "[FSDP()][5/N] Refactor `MixedPrecision` dataclass"




[ghstack-poisoned]

Commit:
Update on "[FSDP()][5/N] Refactor `MixedPrecision` dataclass"


This PR is easy. I moved `MixedPrecision` to `common_utils.py` and reworded the docs a bit.

[ghstack-poisoned]

Commit:
Update on "[FSDP()][5/N] Refactor `MixedPrecision` dataclass"


This PR is easy. I moved `MixedPrecision` to `common_utils.py` and reworded the docs a bit.

[ghstack-poisoned]

mrshenli reviewed (approved):
LGTM. Stamp to unblock. Please address test errors.

Commit:
Update on "[FSDP()][5/N] Refactor `MixedPrecision` dataclass"


This PR is easy. I moved `MixedPrecision` to `common_utils.py` and reworded the docs a bit.

[ghstack-poisoned]

Commit:
Update on "[FSDP()][5/N] Refactor `MixedPrecision` dataclass"


This PR is easy. I moved `MixedPrecision` to `common_utils.py` and reworded the docs a bit.

[ghstack-poisoned]

Commit:
Update on "[FSDP()][5/N] Refactor `MixedPrecision` dataclass"


This PR is easy. I moved `MixedPrecision` to `common_utils.py` and reworded the docs a bit.

[ghstack-poisoned]

Commit:
Update on "[FSDP()][5/N] Refactor `MixedPrecision` dataclass"


This PR is easy. I moved `MixedPrecision` to `common_utils.py` and reworded the docs a bit.

[ghstack-poisoned]

Commit:
Update on "[FSDP()][5/N] Refactor `MixedPrecision` dataclass"


This PR is easy. I moved `MixedPrecision` to `common_utils.py` and reworded the docs a bit.

[ghstack-poisoned]

awgu commented:
Squashed into previous PR.
